etc/docker/kernel-spark-py/Dockerfile

```diff
@@ -23,7 +23,9 @@ RUN apt-get update && \
     libssl-dev \
     && rm -rf /var/lib/apt/lists/*
 
-ENV JAVA_HOME /usr/lib/jvm/java-1.8.0-openjdk-amd64
+RUN mkdir -p /usr/lib/jvm/java
```

Comment:
I'm seeing the following issues with this image and `kernel-spark-r` (from their pod logs):
```
+ CMD=("$SPARK_HOME/bin/spark-submit" --conf "spark.driver.bindAddress=$SPARK_DRIVER_BIND_ADDRESS" --deploy-mode client "$@")
+ exec /usr/bin/tini -g -- /opt/spark/bin/spark-submit --conf spark.driver.bindAddress=10.42.0.247 --deploy-mode client --properties-file /opt/spark/conf/spark.properties --class org.apache.spark.deploy.PythonRunner local:///usr/local/bin/kernel-launchers/python/scripts/launch_ipykernel.py --RemoteProcessProxy.kernel-id 2b027ae7-309f-4421-9fba-14c284e9d328 --RemoteProcessProxy.port-range 0..0 --RemoteProcessProxy.response-address 10.42.0.235:8877 --RemoteProcessProxy.public-key MIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQCa/SIFcGS1qe/LiszCrcpg+IEmS9P0cml/7HBZZuqqARQFXSo74r6MzUCCFDgao5x9fESoVcGZWWraNY8kvGcRe5IJo3Y7rsQ/PVosQloxNcfQtUHZn4F6kEHycr8QwoM0TJW0H9TKttAT/WZ7u0UAIy8YrDoOCe4XYN5Cl79keQIDAQAB --RemoteProcessProxy.spark-context-initialization-mode lazy
/opt/spark/bin/spark-class: line 71: /usr/lib/jvm/java/bin/java: No such file or directory
/opt/spark/bin/spark-class: line 96: CMD: bad array subscript
```

Looking within the images I see:
```
jovyan@c7778786e3ee:/usr/lib/jvm$ ls -l
total 8
drwxr-xr-x 1 root root 4096 Nov 10 23:25 java
lrwxrwxrwx 1 root root   20 Oct 25 17:26 java-1.8.0-openjdk-amd64 -> java-8-openjdk-amd64
drwxr-xr-x 7 root root 4096 Nov 10 23:25 java-8-openjdk-amd64
jovyan@c7778786e3ee:/usr/lib/jvm$ ls java
java-8-openjdk-amd64
jovyan@c7778786e3ee:/usr/lib/jvm$ ls java/java-8-openjdk-amd64
ASSEMBLY_EXCEPTION  bin  docs  include  jre  lib  man  src.zip  THIRD_PARTY_README
```
and wonder if this `mkdir` should be something like:
```suggestion
RUN ln -s /usr/lib/jvm/java-8-openjdk-amd64 /usr/lib/jvm/java
```

Reply:
ok, I think I see what this statement is about: `RUN ln -s $(readlink -f /usr/bin/javac | sed "s:/bin/javac::") ${JAVA_HOME}`

It essentially provides a means of linking the arch-specific directory to a non-arch name and not need to know the architecture type involved in that.  As a result, would simply _removing_ the `mkdir` statement be sufficient?

Reply:
I've confirmed that only removing the `mkdir` statement is sufficient.
```suggestion
```